plugins/scaffolder-backend/src/lib/templating/SecureTemplater.ts

```diff
@@ -64,6 +64,16 @@ const { render, renderCompat } = (() => {
     }
   }
 
+  if (typeof additionalTemplateGlobals !== 'undefined') {
+    for (const [globalName, global] of Object.entries(additionalTemplateGlobals)) {
+      if (typeof global === 'function') {
+        env.addGlobal(globalName, (...args) => JSON.parse(global(...args)));
+      } else {
+        env.addGlobal(globalName, global);
```

Comment:
We'll want to still `JSON.parse` here and stringify from the outside, just to make sure that we're not passing through anything that could be used to escape the sandbox. It's an extra layer of carefulness on top of `vm2`

Reply:
Makes sense thank you for the explanation!